private/bufpkg/bufwkt/cmd/wkt-go-data/main.go

```diff
@@ -0,0 +1,321 @@
+	}
+}
+
+type flags struct {
+	Pkg string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(
+		&f.Pkg,
+		pkgFlagName,
+		"",
+		"The name of the generated package.",
+	)
+}
+
+func run(ctx context.Context, container appflag.Container, flags *flags) error {
+	dirPath := container.Arg(0)
+	packageName := flags.Pkg
+	if packageName == "" {
+		packageName = filepath.Base(dirPath)
+	}
+	readWriteBucket, err := storageos.NewProvider(storageos.ProviderWithSymlinks()).NewReadWriteBucket(dirPath)
+	if err != nil {
+		return err
+	}
+	pathToData, err := getPathToData(ctx, readWriteBucket)
+	if err != nil {
+		return err
+	}
+	protosourceFiles, err := getProtosourceFiles(ctx, container, readWriteBucket)
+	if err != nil {
+		return err
+	}
+	fullNameToMessage, err := protosource.FullNameToMessage(protosourceFiles...)
+	if err != nil {
+		return err
+	}
+	fullNameToEnum, err := protosource.FullNameToEnum(protosourceFiles...)
+	if err != nil {
+		return err
+	}
+	golangFileData, err := getGolangFileData(
+		pathToData,
+		fullNameToMessage,
+		fullNameToEnum,
+		packageName,
+	)
+	if err != nil {
+		return err
+	}
+	_, err = container.Stdout().Write(golangFileData)
+	return err
+}
+
+func getPathToData(ctx context.Context, bucket storage.ReadBucket) (map[string][]byte, error) {
+	pathToData := make(map[string][]byte)
+	if err := storage.WalkReadObjects(
+		ctx,
+		bucket,
+		"",
+		func(readObject storage.ReadObject) error {
+			data, err := io.ReadAll(readObject)
+			if err != nil {
+				return err
+			}
+			pathToData[readObject.Path()] = data
+			return nil
+		},
+	); err != nil {
+		return nil, err
+	}
+	return pathToData, nil
+}
+
+func getProtosourceFiles(
+	ctx context.Context,
+	container appflag.Container,
+	bucket storage.ReadBucket,
+) ([]protosource.File, error) {
+	// TODO: why is this not working? this is the path that should be used, not NewModuleForBucket
+	//module, err := bufmodulebuild.NewModuleBucketBuilder(container.Logger()).BuildForBucket(
+	//ctx,
+	//bucket,
+	//&bufmoduleconfig.Config{},
+	//)
+	module, err := bufmodule.NewModuleForBucket(ctx, bucket)
+	if err != nil {
+		return nil, err
+	}
+	moduleFileSet, err := bufmodulebuild.NewModuleFileSetBuilder(
+		container.Logger(),
+		bufmodule.NewNopModuleReader(),
+	).Build(
+		ctx,
+		module,
+	)
+	if err != nil {
+		return nil, err
+	}
+	image, fileAnnotations, err := bufimagebuild.NewBuilder(container.Logger()).Build(
+		ctx, moduleFileSet,
+		bufimagebuild.WithExcludeSourceCodeInfo(),
+	)
+	if err != nil {
+		return nil, err
+	}
+	if len(fileAnnotations) > 0 {
+		// stderr since we do output to stdouo
+		if err := bufanalysis.PrintFileAnnotations(
+			container.Stderr(),
+			fileAnnotations,
+			"text",
+		); err != nil {
+			return nil, err
+		}
+		return nil, app.NewError(100, "")
+	}
+	return protosource.NewFilesUnstable(ctx, bufimageutil.NewInputFiles(image.Files())...)
+}
+
+func getGolangFileData(
```

Comment:
This seems like an ideal use for a template. I think that would be much more readable.

Reply:
Personally I don't like text templates too much. As soon as there's any sort of control flow that needs to happen it becomes too complicated for me. 

I've changed all the `buffer.WriteString(` for an anonymous function `p(` which is similar to how it's done with other go code generators i've seen and it makes it a lot more readable than `buffer.WriteString`

Reply:
> As soon as there's any sort of control flow that needs to happen it becomes too complicated for me.

I agree 100%... which is why this is the ideal use for one since it's just emitting a hard-coded map 😝 

(Not a blocking comment... just my 2 cents on readability for this function.)

Reply:
For context, this is mostly copy/pasted from `storage-go-data`, didn't want to spend time reinventing the wheel.